csrc/python_frontend/fusion_definition.cpp

```diff
@@ -455,40 +454,40 @@ std::vector<DistributedTensor> FusionDefinition::execute(
 
   // Convert `at::Tensor`s to `DistributedTensor`s.
   std::vector<DistributedTensor> out_dtensors;
-  out_dtensors.reserve(out_tensors.size());
   if (user_sched == nullptr) {
     FusionKernelRuntime* runtime =
         scheds->auto_gen_schedules->getMostRecentKernelRuntime();
     Fusion* fusion = runtime->fusionSegments()->completeFusion();
 
     int64_t tensor_index = 0;
-    for (Val* out_val : fusion->outputs()) {
+    for (auto out_val : fusion->outputs()) {
+      NVF_ERROR(
+          out_val->isA<TensorView>(),
+          "Non tensor outputs not supported currently due to lack of support of DistributedTensor in KernelArgumentHolder");
       auto* out_tv = out_val->as<TensorView>();
       if (fusion->getOutputAlias(out_tv).hide_output) {
         continue;
       }
-
-      const at::Tensor& out_tensor = out_tensors.at(tensor_index);
-      tensor_index++;
+      const at::Tensor& out_tensor = outputs[tensor_index++].as<at::Tensor>();
       const DeviceMesh& mesh = out_tv->getDeviceMesh();
-      DistributedTensor& out_dtensor =
-          out_dtensors.emplace_back(out_tensor, mesh);
+      out_dtensors.emplace_back(DistributedTensor(out_tensor, mesh));
```

Comment:
I prefer the old code. It constructs a DistributedTensor in place which is slightly faster: https://abseil.io/tips/112. Also emplace_back returns a reference to the back element that can be used later.

Reply:
Sure, I can revert to reserver and construction in place, looks mostly like I left this from attempting to support DTensors in kernel argument holder, which didn't go well and I abandoned for now anyways.